### PR TITLE
update convertMap, convertList, and convertSet to match proxygen

### PR DESCRIPTION
**Pull request by shaunrader**
These methods should probably have been updated at the same time as their proxygen counterparts with [this commit](https://github.com/vert-x3/vertx-service-proxy/commit/317a24229fc9465d4cb46937501a2636101330cd)

This will fix a specific problem we were having in production (intermittently and only when clustered) where a `List<JsonObject>` sent from one node to another ended up as `List<LinkedHashMap>`. When trying to access an item in this list via `get(0)` this exception would be thrown due to a failed implicit conversion: `java.util.LinkedHashMap cannot be cast to io.vertx.core.json.JsonObject`.

The previous `convertList` did not account for the `JsonObject` getting converted to a `LinkedHashMap` over the wire when clustered.

### CODE REVIEW

**Comment by cescoffier** [comment]
Could you add a few tests reproducing the bug and checking the fix. There are a set of clustered tests made to reproduce such kind of issue.

**Comment by shaunrader** [comment]
I'll take a look at adding tests hopefully today.

**Comment by shaunrader** [comment]
I added code to the `ServiceProvider` that tries to access a value in the List which caused the existing `testWithListOfJsonObject` test to fail without this fix. It exhibits the same issue we are seeing with a failed cast from `LinkedHashMap` to `JsonObject`.

I don't see `convertMap` or `convertSet` being used in either the generated eb or handler proxies so I couldn't find a way to test them. Is it possible they can be remove altogether?

**Comment by cescoffier** [comment]
I will have a look.

**Comment by eodgooch** [comment]
Can we get this merged? We are currently using a custom jar and would like to get back on the main path.

File: src/test/java/io/vertx/serviceproxy/clustered/ServiceProvider.java [hunk 1]
```diff
@@ -70,12 +70,21 @@ public Service methodWithDataObject(TestDataObject data, Handler<AsyncResult<Tes
 
   @Override
   public Service methodWithListOfDataObject(List<TestDataObject> list, Handler<AsyncResult<List<TestDataObject>>> result) {
+    // access the data
+    if(list != null && list.size() > 0) {
+      TestDataObject data = list.get(0);
+    }
```
Comment (cescoffier) [inline src/test/java/io/vertx/serviceproxy/clustered/ServiceProvider.java]: We should do something with the data.

File: src/test/java/io/vertx/serviceproxy/clustered/ServiceProvider.java [hunk 2]
```diff
@@ -70,12 +70,21 @@ public Service methodWithDataObject(TestDataObject data, Handler<AsyncResult<Tes
...
     result.handle(Future.succeededFuture(list));
     return this;
   }
 
   @Override
   public Service methodWithListOfJsonObject(List<JsonObject> list, Handler<AsyncResult<List<JsonObject>>> result) {
+    // access the data
+    if(list != null && list.size() > 0) {
+      JsonObject data = list.get(0);
```
Comment (cescoffier) [inline src/test/java/io/vertx/serviceproxy/clustered/ServiceProvider.java]: We should do something with data.

**Comment by cescoffier** [comment]
Some test checking nested structure would be great (list of list, map of map, map of list...)

**Comment by shaunrader** [comment]
Added some code to do something with the data in the lists that would previously fail when calling `.get(0)` (as opposed to just getting them).

Re: tests checking nested structures if I'm understanding correctly the vertx-codegen doesn't allow us to create services with `List<List<JsonObject>>` as an input. It fails with: 

`Error:(66, 11) java: Could not generate model for methodWithListOfListOfDataObject(java.util.List<java.util.List<io.vertx.core.json.JsonObject>>,io.vertx.core.Handler<io.vertx.core.AsyncResult<java.util.List<java.util.List<io.vertx.core.json.JsonObject>>>>): type java.util.List<java.util.List<io.vertx.core.json.JsonObject>> is not legal for use for a parameter in code generation`

Or am I misunderstanding your suggestion? Thanks.

**Pull request closed without merging**